packages/block-library/src/embed/embed-bottom-sheet.native.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	LinkSettingsNavigation,
+	FooterMessageLink,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { isURL } from '@wordpress/url';
+import { useDispatch } from '@wordpress/data';
+import { store as noticesStore } from '@wordpress/notices';
+
+const linkSettingsOptions = {
+	url: {
+		label: __( 'Embed link' ),
+		placeholder: __( 'Add link' ),
+		autoFocus: true,
+		autoFill: true,
+	},
+	footer: {
+		label: (
+			<FooterMessageLink
+				href={ __( 'https://wordpress.org/support/article/embeds/' ) }
+				value={ __( 'Learn more about embeds' ) }
+			/>
+		),
+		separatorType: 'topFullWidth',
+	},
+};
+
+const EmbedBottomSheet = ( { value, onClose } ) => {
+	const [ isVisible, setIsVisible ] = useState( true );
+	const [ url, setUrl ] = useState( value );
+	const { createErrorNotice } = useDispatch( noticesStore );
+
+	function onCloseWithUrl() {
```

Comment:
@ceyhun I am noticing that when the bottom sheet is opened and then closed the error below is shown while the Embed link is empty. Is that the expected behavior or do we want to check if the URL is empty or not before checking `isURL`?

Reply:
I decided to change the callback we use for returning the URL to `setAttributes` because `onClose` was being called multiple times and each time has a different URL value:

Example URL: `wordpress.com`
- 1st call: `url = wordpress.com`
- 2nd call: `url = http://wordpress.com`

**Related commit: https://github.com/WordPress/gutenberg/pull/31343/commits/2db674c7c59aeb3b54e14a1930a7a18689635682**